pkg/encrypt/server-side.go

```diff
@@ -90,6 +95,14 @@ type ServerSide interface {
 // Using SSE-S3 the server will encrypt the object with server-managed keys.
 func NewSSE() ServerSide { return s3{} }
 
+// NewSSEKMS returns a new server-side-encryption using SSE-KMS and the provided Key Id.
+func NewSSEKMS(keyID string) ServerSide { return kms{key: keyID, context: nil} }
+
+// NewSSEKMSWithContext returns a new server-side-encryption using SSE-KMS and the provided Key Id and context.
+func NewSSEKMSWithContext(keyID string, context *string) ServerSide {
```

Comment:
Since::
>The encryption context can be any value that you want, provided that the header adheres to the Base64-encoded JSON format. 

I think we can follow this approach:
```
NewSSEKMS(keyID string, context interface{}) (ServerSide, error) {
      // if context is nil -> no context
      // if context implements json.Marshaler -> marshal
      // json.Marshal(context) 
}
```

Reply:
I like this approach and then we don't need the `contextPresent` proposed in the comment below. I'll do this changes tomorrow.